### ISSUE TITLE
Binary uploads converted to string before reaching write stream

### ISSUE THREAD

**Issue by crash83k** [stewarttylerr/ftpserver#8]
I've been trying to figure out where the data is being converted to a string as it's uploaded. 

In the STOR command module, I've changed 

```
this.dataSocket.pipe(stream)
```

to 

```
this.dataSocket.on('data', function(data){
        console.log(typeof data, data.constructor.name);
        stream.write(data, 'binary');
      });
```

What I'm getting is that data is a string.
When I was using pipe, (what you had originally) the binary data was converted immediately to string and corrupting the binary uploads.

**Edit:** To be clear, both `stream.write` and `.pipe(stream)` write corrupted files.

My `write()` method in the FS class is nearly identical to yours:

```
write(filePath, append = false) {      
      const filePoint = Path.resolve(__dirname, 'tmp', filePath.substr(1));
      const pass = fs.createWriteStream(filePoint, {flags : !append ? 'w+' : 'a+'});
      pass.on('error',fs.unlink(filePoint));

      return pass;
    }
```

I'm pointing it at a temp folder where the file is confirmed to be writing to.

Seems no matter what, I'm always receiving the data to my write stream as a string. I can't find where this encoding is being done. Am I missing something?

**Comment by stewarttylerr** [comment]
@crash83k The `TYPE` command sets the encoding for the `dataSocket` (ftp/commands/type.js).
This is then passed to the `dataSocket` encoding (ftp/commands/pasv.js).
Try having the client send `TYPE L` then trying to write the file.

**Comment by crash83k** [comment]
I'm using FileZilla, which is set to Binary mode. The only way I was able to get the encoding to work was to hard code the 'binary' encoding for the dataSocket. 

I've gone through all the options in FileZilla and couldn't find anything specific to the encoding of transfers other than the ASCII vs. Binary modes. Not certain there is a way to force FileZilla to send `TYPE L` manually. 

I also tested with the FTP CLI in Ubuntu and had no luck. It was a very simple test (didn't send `TYPE L`) though. I did the usual basic upload: 
> FTP 
> open 127.0.0.1 7000 (I'm running the server on port 7000)
> PUT a_binary_video_file.mp4

... It puts this in PASV mode automatically and uploads the file.

When it completes, the file is both corrupt and FAR larger than the original.

After changing the `dataSocket.setEncoding()` to `binary`, it was working perfectly.
As an aside, I will only be allowing binary files uploaded through this system as is, so this may work fine for me in the meantime. Also, the users that will actually be doing the uploading are not very computer literate. They understand how to open an FTP client, upload a file, and that's about it. LOL So I won't be able to give them much instruction as to how to use the FTP server or their client.
